### PR TITLE
Update sync logic to deal with repov3 correctly

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

File: indexer/indexer.go [hunk 1]
```diff
@@ -194,6 +189,295 @@ func (ix *Indexer) handleRepoOp(ctx context.Context, evt *repomgr.RepoEvent, op
 	return nil
 }
 
+func (ix *Indexer) crawlAtUriRef(ctx context.Context, uri string) error {
```
Comment (whyrusleeping) [inline indexer/indexer.go:192]: i just moved all this code to the bottom of the file. I want to delete it, but its used by the PDS still so we should probably just extract it to something pds specific (or just rewrite the pds logic better)

File: repo/repo.go [hunk 2]
```diff
@@ -253,18 +253,18 @@ func (r *Repo) Truncate() {
 }
 
 // creates and writes a new SignedCommit for this repo, with `prev` pointing to old value
-func (r *Repo) Commit(ctx context.Context, signer func(context.Context, string, []byte) ([]byte, error)) (cid.Cid, error) {
+func (r *Repo) Commit(ctx context.Context, signer func(context.Context, string, []byte) ([]byte, error)) (cid.Cid, string, error) {
```
Comment (whyrusleeping) [inline repo/repo.go:256]: we might be able to get away with removing the cid from the return value here

**Review by ericvolp12 — APPROVED** [review]
LGTM

**Comment by whyrusleeping** [comment]
lovely net-negative PR

**Pull request merged**